### ISSUE TITLE
Running on a git tag errors

### ISSUE THREAD

**Issue by dherges** [greenkeeperio/greenkeeper-lockfile#30]
On Circle CI, when CI checks out a tag and runs the build from the git tag:

```
greenkeeper-lockfile-update
fatal: ambiguous argument 'undefined': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
/home/ubuntu/.config/yarn/global/node_modules/greenkeeper-lockfile/update.js:35
  if (!info.branchName.startsWith(config.branchPrefix)) {
                      ^

TypeError: Cannot read property 'startsWith' of undefined
    at Module.update [as exports] (/home/ubuntu/.config/yarn/global/node_modules/greenkeeper-lockfile/update.js:35:23)
    at Object.<anonymous> (/home/ubuntu/.config/yarn/global/node_modules/greenkeeper-lockfile/update.js:56:37)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3

greenkeeper-lockfile-update returned exit code 1

Action failed: greenkeeper-lockfile-update
```

**Comment by notjosh** [comment]
This is also causing a problem for us.

Looking at the other services' docs, it's not clear what other services provide on a tag build (it implies like they'd be `undefined` or `''`, not the tag).

In any case, either being `''` or the tag itself wouldn't have any actual impact on how to trigger `greenkeeper-lockfile` (...unless your tag is called `greenkeeper-something`), so there's no real need to cover every corner case here.

But in the spirit of moving forward, imo we can either:

- `branchName: env.CIRCLE_BRANCH || env.CIRCLE_TAG`
- `branchName: env.CIRCLE_BRANCH || ''`

imo the latter has the fewest risks of side effects.

**Comment by janl** [comment]
Closed via #46
